src/build_function.jl

```diff
@@ -265,20 +265,42 @@ function toexpr(p::SpawnFetch{MultithreadedForm}, st)
     end
 end
 
-function _make_array(rhss::AbstractSparseArray, similarto)
-    arr = map(x->_make_array(x, similarto), rhss)
-    if !(arr isa AbstractSparseArray)
-        _make_array(arr, similarto)
+function nzmap(f, x::Union{SubArray, Base.ReshapedArray, LinearAlgebra.Transpose})
+    Setfield.@set! x.parent = nzmap(f, x.parent)
+    x
+end
+
+function nzmap(f, x::AbstractSparseArray)
+    Setfield.@set! x.nzval = nzmap(f, x.nzval)
+    x
+end
+nzmap(f, x) = map(f, x)
+
+_issparse(x::AbstractArray) = issparse(x)
+_issparse(x::Union{SubArray, Base.ReshapedArray, LinearAlgebra.Transpose}) = issparse(parent(A))
+
+function _make_sparse_array(arr, similarto)
+    if arr isa Union{SubArray, Base.ReshapedArray, LinearAlgebra.Transpose}
+        quote
```

Comment:
Is this branch tested?

Reply:
It should be in the MTK build_function tests with Array{SparseArray}

Reply:
Yeah that's why there was a branch here in the first place. But I'll add more test before merging this.

Reply:
now it definitely is.